pkg/aws/awsclient.go

```diff
@@ -57,17 +75,34 @@ func newAwsClientConfig() *awsClientConfig {
 	return &cfg
 }
 
-func newAwsConfig(ctx context.Context, region string) *aws.Config {
-	cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(region))
+func NewClient(ctx context.Context, vc vault.VaultClient, account string) *awsClient {
+	if len(account) == 0 {
+		util.Log().Fatalw("No AWS account name provided")
+	}
+	result, err := getAccounts(ctx, account)
 	if err != nil {
-		util.Log().Fatalw("Error creating AWS configuration %s", err.Error())
+		util.Log().Fatalw("Error getting AWS account info", "error", err.Error())
+	}
+	accounts := result.GetAwsaccounts_v1()
+	if len(accounts) != 1 {
+		util.Log().Fatalw("Expected one AWS with name", "account", account)
+
+	}
+
+	secret, err := vc.ReadSecret(accounts[0].AutomationToken.GetPath())
+	if err != nil {
+		util.Log().Fatalw("Error reading automation token", "error", err.Error())
+	}
+	awsCfg := newAwsClientConfig()
+
+	aws_access_key_id := secret.Data["aws_access_key_id"].(string)
+	aws_secret_access_key := secret.Data["aws_secret_access_key"].(string)
+	cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(awsCfg.Region), config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(aws_access_key_id, aws_secret_access_key, "")))
```

Comment:
Won't be better to pass the access_key and secret_access_key to this function instead of adding the vault dependency?

Reply:
I'd defer to below's conversation